web/html/src/manager/content-management/list-filters/filter.utils.js

```diff
@@ -23,51 +39,38 @@ export function mapFilterFormToRequest(filterForm: FilterFormType, projectLabel:
       requestForm.criteriaValue =
         `${filterForm.packageName || ""}-${epochName}${filterForm.version || ""}-${filterForm.release || ""}.${filterForm.architecture}`;
     }
-  } else if (filterForm.type === filtersEnum.enum.ERRATUM_PKG_NAME.key) {
-    requestForm.criteriaKey = "package_name";
-    requestForm.criteriaValue = filterForm.criteria;
-  } else if (filterForm.type === filtersEnum.enum.ERRATUM_PKG_LT_EVR.key ||
-             filterForm.type === filtersEnum.enum.ERRATUM_PKG_LE_EVR.key ||
-             filterForm.type === filtersEnum.enum.ERRATUM_PKG_EQ_EVR.key ||
-             filterForm.type === filtersEnum.enum.ERRATUM_PKG_GE_EVR.key ||
-             filterForm.type === filtersEnum.enum.ERRATUM_PKG_GT_EVR.key) {
+  } else if (filterForm.type === clmFilterOptions.PACKAGE_NEVR.key) {
     const epochName = !_isEmpty(filterForm.epoch) ? `${filterForm.epoch}:` : '';
-    requestForm.criteriaKey = "package_nevr"
     requestForm.criteriaValue =
       `${filterForm.packageName || ""} ${epochName}${filterForm.version|| ""}-${filterForm.release|| ""}`;
-  } else if (filterForm.type === filtersEnum.enum.ERRATUM.key) {
-    requestForm.criteriaKey = "advisory_name";
-    requestForm.criteriaValue = filterForm.advisoryName;
-  } else if (filterForm.type === filtersEnum.enum.ERRATUM_BYTYPE.key) {
-    requestForm.criteriaKey = "advisory_type";
-    requestForm.criteriaValue = filterForm.advisoryType;
-  } else if (filterForm.type === filtersEnum.enum.ERRATUM_BYDATE.key) {
-    requestForm.criteriaKey = "issue_date";
-    requestForm.criteriaValue = filterForm.issueDate
-      ? moment(Functions.Utils.dateWithoutTimezone(filterForm.issueDate, localTime)).format("YYYY-MM-DDTHH:mm:ss.SSSZ")
-      :  "";
-  } else if (filterForm.type === filtersEnum.enum.ERRATUM_BYSYNOPSIS.key || filterForm.type === filtersEnum.enum.ERRATUM_BYSYNOPSIS_CONTAINS.key) {
-    requestForm.criteriaKey = "synopsis";
-    requestForm.criteriaValue = filterForm.synopsis;
-  } else {
-    requestForm.criteriaKey = "name";
-    requestForm.criteriaValue = filterForm.criteria;
   }
+
   return requestForm;
 }
 
 export function mapResponseToFilterForm(filtersResponse: Array<FilterServerType> = []): Array<FilterFormType> {
   return filtersResponse.map(filterResponse => {
     let filterForm = {};
     filterForm.id = filterResponse.id;
-    filterForm.name = filterResponse.name;
+    filterForm.filter_name= filterResponse.name;
     filterForm.rule = filterResponse.rule;
     filterForm.matcher = filterResponse.matcher;
     filterForm.projects = filterResponse.projects;
 
-    if(filterResponse.criteriaKey === "nevr") {
-      filterForm.type = filtersEnum.enum.PACKAGE_NEVRA.key;
+    const selectedFilterOption = findClmFilterByKey(filterResponse.criteriaKey);
+    // If we can find a filter option using the CriteriaKey we assume the default behavior
+    if(selectedFilterOption) {
```

Comment:
Similar situation as above: does the whole function make sense when `selectedFilterOption` is falsy?

Reply:
This shouldn't happen unless we have a really ugly bug. I added an else block logging more information to the server, so we can at least be aware what's wrong